backend/routers/battle.js

```diff
@@ -38,6 +39,119 @@ async function getUsersOnPlanet(req, res, next) {
   }
 }
 
+function getCasualtyPercentage(troop, targetTroop, correction = 1) {
+  return (targetTroop.attack / (targetTroop.attack + troop.attack))
+    * ((troop.defence + 100) / (2 * troop.defence + 100))
+    * (0.6 + 0.8 * Math.random())
+    * correction;
+}
+
+function applyCasualty(troop, casualtyPercentage) {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const element of troop.countByLevel) {
+    element.count = Math.floor(element.count * (1 - casualtyPercentage));
+  }
+}
+
+async function battle(req, res, next) {
+  const { _id: attackerId } = req.user;
+  const { targetId } = req.params;
+  try {
+    if (attackerId === targetId) {
+      throw createError(400, 'Cannot attack yourself');
+    }
+    const attacker = await UserModel.findOne({ _id: attackerId });
+    const defender = await UserModel.findOne({ _id: targetId });
+    if (!attacker || !defender) {
+      throw createError(404, 'User not found');
+    }
+    const attackerTroop = await TroopModel.findOne(
+      { owner: attackerId },
+      { _id: false, owner: false },
+    );
+    const defenderTroop = await TroopModel.findOne(
+      { owner: targetId },
+      { _id: false, owner: false },
+    );
+
+    const battleReport = [];
+
+    if (!attackerTroop || !attackerTroop.countByLevel || attackerTroop.hp === 0) {
+      throw createError(400, 'You don\'t have troops for attacking');
+    }
+    if (!defenderTroop || !defenderTroop.countByLevel || defenderTroop.hp === 0) {
+      battleReport.push('No enemy troop in the kingdom\n');
+      res.status(201).send({
+        isWin: true,
+        battleReport,
+        myTroopLeft: attackerTroop.hp,
+        enemyTroopLeft: defenderTroop.hp,
+      });
+    }
+
+    // Fleet battle stage
+    const atkCasualtyFleetBattle = getCasualtyPercentage(attackerTroop, defenderTroop);
+    const defCasualtyFleetBattle = getCasualtyPercentage(defenderTroop, attackerTroop);
+    applyCasualty(attackerTroop, atkCasualtyFleetBattle);
+    applyCasualty(defenderTroop, defCasualtyFleetBattle);
+    const fleetBattleWin = atkCasualtyFleetBattle < defCasualtyFleetBattle;
```

Comment:
```suggestion
    const fleetBattleWin = atkCasualtyFleetBattle < defCasualtyFleetBattle;
```

the value will be a boolean, so this means if the user has won or not? because then it should be named something like `isWinner`.